core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +38,28 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store sequentially via {@link #write(MarshalledEntry)}
```

Comment:
Good spot! I'll update the updater to use batches and perform coalescing. It makes sense for the coalescing to be done in memory opposed to redundant write operations on the store.